src/zenml/cli/stack_recipes.py

```diff
@@ -0,0 +1,1016 @@
+import sys
+from pathlib import Path
+from typing import Any, List, Optional, cast
+
+import click
+import python_terraform  # type: ignore
+from packaging.version import Version, parse
+from rich.markdown import Markdown
+from rich.text import Text
+
+from zenml.cli import utils as cli_utils
+from zenml.cli.cli import TagGroup
+from zenml.cli.stack import import_stack, stack
+from zenml.console import console
+from zenml.exceptions import GitNotFoundError
+from zenml.io import fileio
+from zenml.logger import get_logger
+from zenml.utils import io_utils
+from zenml.utils.analytics_utils import AnalyticsEvent, track_event
+
+logger = get_logger(__name__)
+
+EXCLUDED_RECIPE_DIRS = [""]
+STACK_RECIPES_GITHUB_REPO = "https://github.com/zenml-io/mlops-stacks.git"
+STACK_RECIPES_REPO_DIR = "zenml_stack_recipes"
+VARIABLES_FILE = "values.tfvars.json"
+
+
+class Terraform:
+    """Class to represent terraform applications."""
+
+    def __init__(self, path: str) -> None:
+        """Creates a client that can be used to call terraform commands.
+
+        Args:
+            path: the path to the stack recipe.
+        """
+        self.tf = python_terraform.Terraform(working_dir=str(path))
+
+    def check_installation(self) -> None:
+        """Checks if terraform is installed on the host system.
+
+        Raises:
+            RuntimeError: if terraform is not installed.
+        """
+        if not self._is_installed():
+            raise RuntimeError(
+                "Terraform is required for stack recipes to run and was not "
+                "found installed on your machine. Please visit "
+                "https://learn.hashicorp.com/tutorials/terraform/install-cli "
+                "to install it."
+            )
+
+    def _is_installed(self) -> bool:
+        """Checks if terraform is installed on the host system.
+
+        Returns:
+            True if terraform is installed, false otherwise.
+        """
+        # check terraform version to verify installation.
+        ret_code, _, _ = self.tf.cmd("-version")
+        return bool(ret_code == 0)
+
+    def apply(self) -> str:
+        """Function to call terraform init and terraform apply.
+
+        The init call is not repeated if any successful execution has
+        happened already, to save time.
+
+        Returns:
+            The path to the stack YAML configuration file as a string.
+
+        Raises:
+            RuntimeError: if terraform init fails.
+        """
+        # this directory gets created after a successful init
+        previous_run_dir = os.path.join(self.tf.working_dir, ".ignoreme")
+        if fileio.exists(previous_run_dir):
+            logger.info(
+                "Terraform already initialized, "
+                "terraform init will not be executed."
+            )
+        else:
+            ret_code, _, _ = self.tf.init(capture_output=False)
+            if ret_code != 0:
+                raise RuntimeError("The command 'terraform init' failed.")
```

Comment:
```suggestion
                raise RuntimeError("The command 'terraform init' failed.")
```

Is there any way we could tell the user what to do to make it work?

Reply:
Yes, I agree. I catch this error in the `deploy` and `destroy` CLI functions and have added extensive instructions there on potential actions to take.